lib/models/blueprint.js

```diff
@@ -770,7 +864,45 @@ let Blueprint = CoreObject.extend({
 
     this._ignoreUpdateFiles();
 
-    return finishProcessingForUninstall(fileInfos.filter(isValidFile));
+    fileInfos = fileInfos.filter(isValidFile).reduce((acc, info) => {
+      // if it's possible that this blueprint could have produced either typescript OR javascript, we have to do some
+      // work to figure out which files to delete.
+      if (this.shouldTransformTypeScript) {
+        if (this.options.typescript === true) {
```

Comment:
lets make `shouldTransformTypeScript === false && options.typescript === true` a nice error (using `SilentError`) for the user (telling them that they asked to delete the `--typescript` version of this blueprint, but it doesn't support typescript anyways)

Reply:
There's a possibility that people could have blueprints that are *exclusively* TypeScript, in which case this error message wouldn't end up being accurate.

That said, we're not at any risk of accidentally deleting someone's JS files in this scenario since `--typescript` forces the command to only look for `.ts` files. It'll simply overlook the JS files.